src/magic_folder/magic_folder.py

```diff
@@ -364,14 +397,17 @@ def from_config(cls, client_node, name, config):
                 local_dir_config,
                 base=client_node.config.get_config_path(),
             ),
-            dbfile=abspath_expanduser_unicode(db_filename),
+            db=database,
             umask=config["umask"],
             name=name,
             downloader_delay=poll_interval,
+            clock=reactor,
+            snapshot_service=snapshot_service,
```

Comment:
If the only thing `MagicFolder` does with `snapshot_service` is add it as a child of itself, maybe this caller should just add it as a child and `MagicFolder` can be unaware of it?

Reply:
It probably WILL want to do other things (e.g. "status API" will want access to it -- or at least to the creator-of-local-things queue). I'm kind of unclear what the right pattern for "a bunch of `IService` things really is, though; maybe we can gain access by asking for the service by name? Or put the arg back as above if/when we actually-do need access?

I don't have any strong feelings on where the "setServiceParent" call goes ..

Reply:
I think removing it and putting it back if we need it is the right path.  We don't know exactly what we'll need.  Better to leave the field wide open for when someone actually sits down to figure it out.